modules/programs/process-compose.nix

```
@@ -0,0 +1,42 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib) mkIf;
+  cfg = config.programs.process-compose;
+  settingsFormat = pkgs.formats.yaml { };
+in
+{
+  meta.maintainers = [ lib.hm.maintainers.bbigras ];
+  options.programs.process-compose = {
+    enable = lib.mkEnableOption "the process-compose orchestrator";
+    package = lib.mkPackageOption pkgs "process-compose" { nullable = true; };
+    settings = lib.mkOption {
+      type = settingsFormat.type;
+      default = { };
+      description = "The process-compose configuration.";
+    };
+  };
+  config =
+    let
+      configDir =
+        if (pkgs.stdenv.targetPlatform.isDarwin) then
+          "Library/Application Support/process-compose"
+        else
+          "${config.xdg.configHome}/process-compose";
+    in
+    mkIf cfg.enable {
+      home = {
+        packages = mkIf (cfg.package != null) [ cfg.package ];
+
+        file.process-compose-settings = {
+          enable = cfg.settings != null && cfg.settings != { };
```

settings can't be null here
```suggestion
          enable = cfg.settings != { };
```